server/admin_management/tests/services/test_user_service.py

```diff
@@ -44,3 +50,101 @@ def test_find_or_create(user_service: UserService):
     users = user_service.get_users()
     # verify no user created
     assert len(users) == len(after_add_users_count)
+
+    # give the existing user with a new username need to be updated
+    updated_user = user_service.find_or_create(
+        TEST_NEW_IDIR_USER.user_type_code,
+        NEW_USERNAME,
+        TEST_NEW_IDIR_USER.user_guid,
+        TEST_NEW_IDIR_USER.create_user,
+    )
+    users = user_service.get_users()
+    # verify no user created
+    assert len(users) == len(after_add_users_count)
+    # verify username is updated
+    assert updated_user.user_name == NEW_USERNAME
+
+
+def test_update_user_name(user_service: UserService, user_repo: UserRepository):
+    # create a user
+    new_user = user_repo.create_user(TEST_NEW_IDIR_USER)
+    # verify new user is created
+    found_user = user_service.get_user_by_domain_and_name(
+        TEST_NEW_IDIR_USER.user_type_code, TEST_NEW_IDIR_USER.user_name
+    )
+    assert new_user.user_id == found_user.user_id
+
+    # test update user name
+    updated_user = user_service.update_user_name(
+        found_user, NEW_USERNAME, TEST_NEW_IDIR_USER.create_user
+    )
+    assert updated_user.user_name == NEW_USERNAME
+    # verify the username is updated
+    found_user = user_service.get_user_by_domain_and_name(
+        TEST_NEW_IDIR_USER.user_type_code, NEW_USERNAME
+    )
+    assert found_user.user_id == updated_user.user_id
+    assert found_user.user_name == NEW_USERNAME
+
+    # test user name no need to be updated
+    updated_user = user_service.update_user_name(
+        found_user, NEW_USERNAME.lower(), TEST_NEW_IDIR_USER.create_user
+    )
+    assert updated_user.user_name == NEW_USERNAME
+    # verify the username is not updated
+    found_user = user_service.get_user_by_domain_and_name(
+        TEST_NEW_IDIR_USER.user_type_code, NEW_USERNAME
+    )
+    assert found_user.user_id == updated_user.user_id
+    assert found_user.user_name == NEW_USERNAME
+
+
+def test_update_user_business_guid(
+    user_service: UserService, user_repo: UserRepository
+):
+    # create a business user
+    new_user = user_repo.create_user(TEST_NEW_BCEID_USER)
+    # verify new user is created
+    found_user = user_service.get_user_by_domain_and_name(
+        TEST_NEW_BCEID_USER.user_type_code, TEST_NEW_BCEID_USER.user_name
+    )
+    assert new_user.user_id == found_user.user_id
+
+    # test update business guid when no business guid stored
+    updated_user = user_service.update_user_business_guid(
+        found_user.user_id,
+        TEST_USER_BUSINESS_GUID_BCEID,
+        TEST_NEW_IDIR_USER.create_user,
+    )
+    assert updated_user.business_guid == TEST_USER_BUSINESS_GUID_BCEID
+    # verify the business_guid is updated
+    found_user = user_service.get_user_by_domain_and_name(
+        TEST_NEW_BCEID_USER.user_type_code, TEST_NEW_BCEID_USER.user_name
+    )
```

Comment:
Not sure if I that's what I thought, maybe you don't even need to do `found_user = user_service.get_user_by_domain_and_name` because of line 108 is the same query to the database...
So if you remove line 121 to 123, probably the below assert (124/125) might still be working.
However, just wondering, no need to change any code....

Reply:
yeah you're right, it's the same, just to make sure if we try to find again, it returns the same